packages/ui/src/validators/modals/SetNomineesModal/SetNomineesModal.tsx

```diff
@@ -0,0 +1,306 @@
+import { useMachine } from '@/common/hooks/useMachine'
+import { useModal } from '@/common/hooks/useModal'
+import { useObservable } from '@/common/hooks/useObservable'
+import { useSignAndSendTransaction } from '@/common/hooks/useSignAndSendTransaction'
+import { transactionMachine } from '@/common/model/machines'
+import { useStakingTransactions } from '@/validators/hooks/useStakingSDK'
+
+import { SetNomineesModalCall } from '.'
+
+export const SetNomineesModal = () => {
+  const { modalData } = useModal<SetNomineesModalCall>()
+
+  if (!modalData) {
+    return null
+  }
+
+  return <SetNomineesModalInner stash={modalData.stash} currentNominations={modalData.nominations} />
+}
+
+interface SetNomineesModalInnerProps {
+  stash: string
+  currentNominations: string[]
+}
+
+const SetNomineesModalInner = ({ stash, currentNominations }: SetNomineesModalInnerProps) => {
+  const { hideModal } = useModal<SetNomineesModalCall>()
+  const { api } = useApi()
+  const { allAccounts } = useMyAccounts()
+  const { nominate } = useStakingTransactions()
+  const [state, , service] = useMachine(transactionMachine)
+
+  const [selectedValidators, setSelectedValidators] = useState<string[]>(currentNominations)
+  const [selectedValidatorToAdd, setSelectedValidatorToAdd] = useState<string | null>(null)
+
+  // Get all validators from the chain
+  const allValidators = useObservable(() => {
+    if (!api) return of([] as string[])
+    return api.query.session.validators().pipe(
+      map((validators) => validators.map((v) => v.toString())),
+      first()
+    )
+  }, [api?.isConnected])
+
+  // Get validator preferences (commission) for all validators
+  const validatorsWithDetails = useObservable(() => {
+    if (!api || !allValidators || allValidators.length === 0)
+      return of([] as Array<{ account: string; commission?: number }>)
+
+    return combineLatest(
+      allValidators.map((validator) =>
+        api.query.staking.validators(validator).pipe(
+          map((prefs: any) => {
+            if (prefs.isEmpty) return { account: validator, commission: undefined }
+            const prefsData = prefs.unwrap ? prefs.unwrap() : prefs
+            return {
+              account: validator,
+              commission: prefsData.commission ? prefsData.commission.toNumber() / 10_000_000 : undefined,
+            }
+          }),
+          first()
+        )
+      )
+    )
+  }, [api?.isConnected, allValidators])
+
+  const availableValidators = validatorsWithDetails || []
+  const isLoadingValidators = allValidators === undefined || validatorsWithDetails === undefined
+
+  // Get validator options for the select dropdown
+  const validatorOptions = useMemo(() => {
+    return availableValidators.map((v) => {
+      const address = encodeAddress(v.account)
+      const commission = v.commission !== undefined ? ` (${v.commission.toFixed(2)}%)` : ''
+      return `${address}${commission}`
+    })
+  }, [availableValidators])
```

Comment:
Already selected validators should be removed from `validatorOptions`. Try if this works:
```suggestion
  const validatorOptions = useMemo(() => {
    return availableValidators
      .filter((v) => !selectedValidators.includes(encodeAddress(v.account)))
      .map((v) => {
        const address = encodeAddress(v.account)
        const commission = v.commission !== undefined ? ` (${v.commission.toFixed(2)}%)` : ''
        return `${address}${commission}`
    })
  }, [availableValidators, selectedValidators])
```